docs/js/features/odata/use-generated-client.md

```diff
@@ -0,0 +1,305 @@
+    - for `getAll`, they are mainly OData query parameters:
+        - [`select`](#Select) restricts the response to the given selection of properties in the request. (also used by getByKey)
+        - [`filter`](#Filter) Add filter statements to the request.
+        - `orderBy`
+        - `skip` skips number of entities. Useful for paging.
+        - `top` limits number of returned entities.
+- _myDestination_ is an URL of the server, it can be declared as
+```ts
+var myDestination : {url: '<yourURL>'}
+```
+
+Below different OData features are documented using the [Business Partner Service](https://api.sap.com/api/API_BUSINESS_PARTNER/resource) on S/4HANA as an example.
+<!--The following code snippets assume that an instance of this service is set up:
+
+```ts
+constructor(private readonly businessPartnerService: BusinessPartnerService) {}
+```
+
+:::tip Structure your code
+All of the following code can be organised in service files, e.g., `MyEntity.service.ts`
+:::
+-->
+
+## OData Features
+
+### Basic CRUD Operations
+
+Create, Read, Update and Delete operations on entities are usually built inside the service class:
+
+```ts
+BusinessPartnerAddress.requestBuilder().create(partner);
+BusinessPartnerAddress.requestBuilder().getAll();
+BusinessPartnerAddress.requestBuilder().getByKey('id');
+BusinessPartnerAddress.requestBuilder().update(partner);
+BusinessPartnerAddress.requestBuilder().delete();
+```
+
+Each of the above statements returns a builder object that allows for specifying certain request parameters, depending on the operation.
+
+<!--Maybe the Query parameters and Request parameters are redundant because already explained previously?-->
+
+The following query parameters and request options are available for these operations:
+
+Query parameters:
+- `$select` and `$expand` are available on reading a single or multiple entities
+- `$filter`, `$top`, `$skip` and `$orderby` are available only when reading a collection of entities
+
+Request parameters:
+- Update and delete operations allow to modify how ETags are handled:
+   - By default an ETag is send if one is present on the entity being modified.
+   - `ignoreVersionIdentifier()` will instead always send a `*` which acts as a wildcard to match all ETags.
+- All operations allow for adding custom headers via `withCustomHeaders(...)`
+
+
+#### Handling of ETags
+
+An [ETag](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/ETag) is a version identifier which is often used to implement an optimistic locking mechanism. The SDK will try to read version identifiers from responses and set them when sending OData requests.
+
+Consider the following example:
+
+```ts
+//the function containing this code must be async
+const partner = BusinessPartner.requestBuilder()
+    .getByKey('id')
+    .execute({
+        url: '<yourURL>'
+    });
+BusinessPartner.requestBuilder()
+    .update(await partner)
+    .execute({
+        url: '<yourURL>'
+    });
```

Comment:
I would remove that comment. Instead you can consider to just wrap it in an async function. 
```suggestion
async function modifyBusinessPartner(id) {
  const destination = { url: 'https://my.s4-system.com' };
  
  const partner = await BusinessPartner.requestBuilder()
      .getByKey(id)
      .execute(destination);
      
  // do some modification
  applyModification(partner);
  
  return BusinessPartner.requestBuilder()
      .update(partner)
      .execute(destination);
}
```